### ISSUE TITLE
Add dbt 1.11+ function() macro to apply_dbt_builtins (Jinja templater)

### ISSUE THREAD

**Issue by Thrasi** [sqlfluff/sqlfluff#7580]
### Search before asking

- [x] I searched the issues and found no similar issues.

### What Happened

When using `apply_dbt_builtins = True` with the Jinja templater, sqlfluff fails to recognise the `function()` Jinja macro introduced in **dbt Core v1.11** for referencing Python UDFs as first-class dbt resources (`function-paths` in dbt-bigquery).

### Expected Behaviour

`apply_dbt_builtins = True` should include a mock for `function()` alongside the existing mocks for `ref()`, `source()`, `config()`, `var()`, and `is_incremental()`, so that files using `{{ function('model_name') }}(args)` lint without errors.

### Observed Behaviour

Linting any `.sql` file that contains `{{ function('some_udf') }}(arg1, arg2)` produces:

```
L:  71 | P:  10 | LT06 | Function name not immediately followed by parenthesis. [layout.functions]
L:  71 | P:  14 |  TMP | Undefined jinja template variable: 'function'
```

The `TMP` error is because `function` is not in `apply_dbt_builtins`. As a side-effect, the undefined variable leaves whitespace between the placeholder and the `(`, triggering `LT06` as an unfixable violation.

Note: This is specific to the **Jinja templater**. The dbt templater delegates compilation to dbt itself, which resolves `function()` natively — no mock is needed there.

### How to reproduce

**`.sqlfluff`:**
```ini
[sqlfluff]
dialect = bigquery
templater = jinja

[sqlfluff:templater:jinja]
apply_dbt_builtins = True
```

**`test.sql`:**
```sql
SELECT {{ function('my_dataset_my_udf') }}(some_column, other_column) AS result
FROM {{ ref('my_model') }}
```

**Command:**
```bash
sqlfluff lint test.sql
```

**Output:**
```
== [test.sql] FAIL
L:   1 | P:  10 | LT06 | Function name not immediately followed by parenthesis. [layout.functions]
L:   1 | P:  14 |  TMP | Undefined jinja template variable: 'function'
```

**Workaround** (adding a manual mock resolves both errors):
```ini
[sqlfluff:templater:jinja:macros]
function = {% macro function(function_name) %}MOCK_FUNCTION{% endmacro %}
```

### Dialect

BigQuery (but the issue applies to any dialect using dbt 1.11+ UDFs with the Jinja templater)

### Version

```
sqlfluff, version 3.5.0
Python 3.13
dbt-bigquery 1.11.x
sqlfluff-templater-dbt 3.5.0
```

Also reproduced on sqlfluff 4.0.4.

### Configuration

```ini
[sqlfluff]
dialect = bigquery
templater = jinja

[sqlfluff:templater:jinja]
apply_dbt_builtins = True
```

### Suggested fix

In `src/sqlfluff/core/templaters/builtins/dbt.py`, add `function` to the `DBT_BUILTINS` dict:

```python
"function": FunctionWrapper("function", lambda function_name: RelationEmulator(function_name)),
```

This renders `{{ function('my_udf') }}(args)` as `my_udf(args)` — a valid parseable SQL function call, consistent with how `ref()` and `source()` are handled.

### Are you willing to submit a PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct
